js/common/lib/tensor-factory.ts

```diff
@@ -165,4 +180,57 @@ export interface TensorFactory {
    */
   fromImage(bitmap: ImageBitmap, options: TensorFromImageBitmapOptions):
       Promise<TypedTensor<'float32'>|TypedTensor<'uint8'>>;
+
+  /**
+   * create a tensor from a WebGL texture
+   *
+   * @param texture - the WebGLTexture object to create tensor from
+   * @param options - An optional object representing options for creating tensor from WebGL texture.
+   *
+   * The options include following properties:
+   * - `width`: the width of the texture. Required.
+   * - `height`: the height of the texture. Required.
+   * - `format`: the format of the texture. If omitted, assume 'RGBA'.
+   * - `download`: an optional function to download the tensor data from GPU to CPU. If omitted, the GPU data
+   * will not be able to download. Usually, this is provided by a GPU backend for the inference outputs. Users don't
+   * need to provide this function.
```

Comment:
`download` and `dispose` is designed for ORT to create outputs tensors, instead of for users to create input tensors. onnxruntime-common does not implement the downloading so `tensor.getData` is actually calling the `download` function passed in.